ddsc/ddsclient.py

```diff
@@ -114,6 +114,30 @@ def fetch_project(self, args, must_exist=True, include_children=False):
             print("Done fetching list of files.".format(project_name_or_id.value))
         return project
 
+    def make_user_list(self, emails, usernames):
+        """
+        Given a list of emails and usernames fetch DukeDS user info.
+        Parameters that are None will be skipped.
+        :param emails: [str]: list of emails (can be null)
+        :param usernames:  [str]: list of usernames(netid)
+        :return: [RemoteUser]: details about any users referenced the two parameters
+        """
+        to_users = []
+        remaining_emails = [] if not emails else list(emails)
+        remaining_usernames = [] if not usernames else list(usernames)
+        for user in self.remote_store.fetch_all_users():
```

Comment:
Fetching all the users from DukeDS to find a handful of users by email seems excessive. I do recognize that this is the only way to find users by username or email using the API, and that this function saves an API call over  calling `lookup_user_by_username()` and `lookup_user_by_email()`.

That said, the list of registered users will continue to grow from the ~300 it is now, so I'll make an issue to request a more targeted API endpoint.

Reply:
Along the same lines looking for project by name has the same type of issue.

Reply:
Ah good point. Just created https://github.com/Duke-Translational-Bioinformatics/duke-data-service/issues/1103